Can't connect to Cassandra 2.1.0

Please answer these questions before submitting your issue. Thanks!

### What version of Cassandra are you using?
2.1.0

### What version of Gocql are you using?
go version go1.11.4 darwin/amd64

### What did you do?
 ```
func init() {

	var err error

	// cluster := gocql.NewCluster("localhost", "127.0.0.2", "127.0.0.3")
	cluster := gocql.NewCluster("127.0.0.1")
	cluster.Keyspace = "user_settings"
	cluster.ProtoVersion = 2
	Session, err = cluster.CreateSession()
	if err != nil {
		fmt.Printf("err is %v\n", err)
		panic(err)
	}
	fmt.Println("cassandra init done")
 }
```

### What did you expect to see?
A successful connection message

### What did you see instead?
```
panic: no valid connect address for host: 
[HostInfo connectAddress="<nil>" peer="<nil>" rpc_address="<nil>" broadcast_address="<nil>" preferred_ip="<nil>" connect_addr="0.0.0.0" connect_addr_source="invalid" port=9042 data_centre="datacenter1" rack="rack1" host_id="265e9938-2e6b-4320-8438-9fcfb5b97a81" version="v2.1.0" state=UP num_tokens=1]. Is your cluster configured correctly?
```
---

If you are having connectivity related issues please share the following additional information

### Describe your Cassandra cluster
please provide the following information

- output of `nodetool status`
```
ccm status
Cluster: 'test'
---------------
node1: UP
node2: UP
node3: UP
```

- output of `SELECT peer, rpc_address FROM system.peers`
```
ccm node1 cqlsh
Connected to test at 127.0.0.1:9042.
[cqlsh 5.0.1 | Cassandra 2.1.0 | CQL spec 3.2.0 | Native protocol v3]
Use HELP for help.
cqlsh> SELECT peer, rpc_address FROM system.peers;

 peer      | rpc_address
-----------+-------------
 127.0.0.3 |   127.0.0.3
 127.0.0.2 |   127.0.0.2

(2 rows)
```
- rebuild your application with the `gocql_debug` tag and post the output

--- a/conn.go
+++ b/conn.go
@@ -165,7 +165,7 @@ func (s *Session) dial(host *HostInfo, cfg *ConnConfig, errorHandler ConnErrorHa
 	port := host.port
 
 	// TODO(zariel): remove these
-	if len(ip) == 0 || ip.IsUnspecified() {
+	if !validIpAddr(ip) {
 		panic(fmt.Sprintf("host missing connect ip address: %v", ip))
 	} else if port == 0 {
 		panic(fmt.Sprintf("host missing port: %v", port))
@@ -1418,7 +1418,7 @@ func (c *Conn) localHostInfo(ctx context.Context) (*HostInfo, error) {
 	port := c.conn.RemoteAddr().(*net.TCPAddr).Port
 
 	// TODO(zariel): avoid doing this here
-	host, err := c.session.hostInfoFromMap(row, port)
+	host, err := c.session.hostInfoFromMap(row, &HostInfo{connectAddress: c.host.connectAddress, port: port})
 	if err != nil {
 		return nil, err
 	}

--- a/host_source.go
+++ b/host_source.go
@@ -450,15 +450,11 @@ func checkSystemSchema(control *controlConn) (bool, error) {
 
 // Given a map that represents a row from either system.local or system.peers
 // return as much information as we can in *HostInfo
-func (s *Session) hostInfoFromMap(row map[string]interface{}, port int) (*HostInfo, error) {
+func (s *Session) hostInfoFromMap(row map[string]interface{}, host *HostInfo) (*HostInfo, error) {
 	const assertErrorMsg = "Assertion failed for %s"
 	var ok bool
 
 	// Default to our connected port if the cluster doesn't have port information
-	host := HostInfo{
-		port: port,
-	}
-
 	for key, value := range row {
 		switch key {
 		case "data_center":
@@ -552,7 +548,7 @@ func (s *Session) hostInfoFromMap(row map[string]interface{}, port int) (*HostIn
 	host.connectAddress = ip
 	host.port = port
 
-	return &host, nil
+	return host, nil
 }
 
 // Ask the control node for host info on all it's known peers
@@ -575,7 +571,7 @@ func (r *ringDescriber) getClusterPeerInfo() ([]*HostInfo, error) {
 
 	for _, row := range rows {
 		// extract all available info about the peer
-		host, err := r.session.hostInfoFromMap(row, r.session.cfg.Port)
+		host, err := r.session.hostInfoFromMap(row, &HostInfo{port: r.session.cfg.Port})
 		if err != nil {
 			return nil, err
 		} else if !isValidPeer(host) {
@@ -637,7 +633,7 @@ func (r *ringDescriber) getHostInfo(ip net.IP, port int) (*HostInfo, error) {
 		}
 
 		for _, row := range rows {
-			h, err := r.session.hostInfoFromMap(row, port)
+			h, err := r.session.hostInfoFromMap(row, &HostInfo{connectAddress: ip, port: port})
 			if err != nil {
 				return nil, err
 			}

